drivers/media/platform/bcm2835/bcm2835-unicam.c

```diff
@@ -2428,35 +2411,33 @@ static int register_node(struct unicam_device *unicam, struct unicam_node *node,
 
 	/* Define the device names */
 	snprintf(vdev->name, sizeof(vdev->name), "%s-%s", UNICAM_MODULE_NAME,
-		 node->pad_id == IMAGE_PAD ? "image" : "embedded");
+		 pad_id == IMAGE_PAD ? "image" : "embedded");
 
 	video_set_drvdata(vdev, node);
-	vdev->entity.flags |= MEDIA_ENT_FL_DEFAULT;
+	if (pad_id == IMAGE_PAD)
+		vdev->entity.flags |= MEDIA_ENT_FL_DEFAULT;
+	node->pad.flags = MEDIA_PAD_FL_SINK;
+	media_entity_pads_init(&vdev->entity, 1, &node->pad);
 
 	node->dummy_buf_cpu_addr = dma_alloc_coherent(&unicam->pdev->dev,
 						      DUMMY_BUF_SIZE,
 						      &node->dummy_buf_dma_addr,
-						      GFP_ATOMIC);
+						      GFP_KERNEL);
 	if (!node->dummy_buf_cpu_addr) {
 		unicam_err(unicam, "Unable to allocate dummy buffer.\n");
 		return -ENOMEM;
 	}
-
-	if (node->pad_id == METADATA_PAD) {
-		v4l2_disable_ioctl(vdev, VIDIOC_DQEVENT);
-		v4l2_disable_ioctl(vdev, VIDIOC_SUBSCRIBE_EVENT);
-		v4l2_disable_ioctl(vdev, VIDIOC_UNSUBSCRIBE_EVENT);
-	}
-	if (node->pad_id == METADATA_PAD ||
+	/* TODO: review which of these disables are still needed */
+	if (pad_id == METADATA_PAD ||
```

Comment:
Is this TODO still needed?

Reply:
Yes.
Hans' comment on https://patchwork.linuxtv.org/patch/63531/

> Some of these are probably no longer needed since the v4l2 core will disable them for you.

I've removed the disables of the event ioctls which we discussed the other day, but I didn't look into these.
I believe it is true that the core will disable some ioctls based on this being a metadata type, but it'd need careful checking, and may be different between 5.4 and linux-media-next.